functions/backup.bash

```diff
@@ -375,30 +369,56 @@ mirror_SD() {
     echo "FAILED (destination mounted)"
     return 1
   fi
+  if [[ ! -d $syncMount ]]; then
+    mkdir -p "$syncMount"
+  fi
+
   if [[ "$1" == "raw" ]]; then
-    echo "Creating a raw partition copy, be prepared this may take long such as 20-30 minutes for a 16 GB SD card"
-    if ! cond_redirect dd if="${src}" bs=1M of="${dest}"; then echo "FAILED (raw device copy)"; dirty="yes"; fi
-    if ! (yes | cond_redirect set-partuuid "${dest}2" random); then echo "FAILED (set random PARTUUID)"; dirty="yes"; fi
+    for i in 1 2; do
+      srcSize="$(blockdev --getsize64 "$src"p${i})"
+      destSize="$(blockdev --getsize64 "$dest"${i})"
+      if [[ "$destSize" -lt "$srcSize" ]]; then
+        echo "FAILED (raw device copy of ${src}${i} larger than ${dest}${i})"
+        return 1
+      fi
+    done
+    echo "Taking a raw partition copy, be prepared this may take long such as 20-30 minutes for a 16 GB SD card"
+    if ! cond_redirect dd if="${src}p1" bs=1M of="${dest}1" status=progress; then echo "FAILED (raw device copy of ${dest}1)"; dirty="yes"; fi
+    if ! cond_redirect dd if="${src}p2" bs=1M of="${dest}2" status=progress; then echo "FAILED (raw device copy of ${dest}2)"; dirty="yes"; fi
+    #origPartUUID=$(blkid "${src}p2" | sed -n 's|^.*PARTUUID="\(\S\+\)".*|\1|p')
+    origPartUUID=$(blkid "${src}p2" | sed -n 's|^.*PARTUUID="\(\S\+\)".*|\1|p' | sed -e 's/-02//g')
+    if ! partUUID=$(yes | cond_redirect set-partuuid "${dest}2" random | awk '/^PARTUUID/ { print substr($7,1,length($7) - 3) }'); then echo "FAILED (set random PARTUUID)"; dirty="yes"; fi
+    if ! cond_redirect tune2fs "${dest}2" -U random; then echo "FAILED (set random UUID)"; dirty="yes"; fi
+    mount "${dest}1" "$syncMount"
+    sed -i "s|${origPartUUID}|${partUUID}|g" "${syncMount}"/cmdline.txt
+    umount "$syncMount"
+    mount "${dest}2" "$syncMount"
+    sed -i "s|${origPartUUID}|${partUUID}|g" "${syncMount}"/etc/fstab
+    sed -i 's|^What=.*|What=/dev/mmcblk0p3|g' "${syncMount}/etc/systemd/system/${storageDir}".mount
```

Comment:
Will this get reset by syncs?

Reply:
This code *is* what is being executed on every sync so it's effectively kept.
That's the storage mount which would point to the (no more existing) external SD after we had to exchange cards.

Reply:
OK, I just misread the code.